src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.ReportGenerator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public List<String> generateReport() {
+        Map<String, Fruit> allFruits = Storage.getAllFruits();
+        return allFruits.values().stream()
+                .map(fruit -> fruit.getName() + "," + fruit.getQuantity())
+                .collect(Collectors.toList());
+    }
```

Comment:
It seems like the generateReport method is returning a List of Strings, which may not be the best representation for a report. Usually, a report is a single String that can be directly written to a file or displayed. Consider changing the return type of this method to String and concatenating the entries with a newline character to form a complete report.